gcp/api/server.py

```diff
@@ -852,16 +849,17 @@ def query_by_version(context: QueryContext,
                                          ecosystem, purl, version)
     bugs.extend(new_bugs)
 
-    new_bugs, _ = yield _query_by_generic_version(context, query, package_name,
-                                                  ecosystem, purl, version)
-    for bug in new_bugs:
-      if bug not in bugs:
-        bugs.append(bug)
+  # Try querying by generic version for all cases.
+  new_bugs, _ = yield _query_by_generic_version(context, query, package_name,
+                                                ecosystem, purl, version)
+  for bug in new_bugs:
+    if bug not in bugs:
+      bugs.append(bug)
 
-    # Trying both is too difficult/ugly with paging
-    # Our documentation states that this is an invalid query
-    # context.service_context.abort(grpc.StatusCode.INVALID_ARGUMENT,
-    #                               'Ecosystem not specified')
+  # Trying both is too difficult/ugly with paging
```

Comment:
Always doing both queries for semver ecosystems would mess up pagination for those requests.
Though I don't really know if we can fix it, and I don't know how many semver packages actually trigger pagination in practice.